src/main/java/edu/fiuba/algo3/modelo/distribuciondepaises/Mapa.java

```diff
@@ -0,0 +1,213 @@
+
+
+
+public class Mapa {
+    private Hashtable<String, Pais> paises;
+    private Hashtable<String, Continente> continentes;
+    private String rutaArchivo;
+
+    public Mapa(String rutaArchivo){
+        this.paises = new Hashtable<String, Pais>();
+        this.continentes = new Hashtable<String, Continente>();
+        this.rutaArchivo = rutaArchivo;
+        this.crearContinentes();
+        try {
+            this.crearPaises();
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void crearPaises() throws FileNotFoundException {
+        FileReader fileReader = new FileReader(rutaArchivo);
+        CSVReader csvReader = new CSVReader(fileReader);
+
+        ArrayList<String> listaLector;
+        try {
+            String[] linea;
+            csvReader.readNext();
+            while ((linea = csvReader.readNext()) != null) {
+                Pais paisAux = new Pais(linea[0]);
+                paises.put(linea[0],paisAux);
+                continentes.get(linea[1]).agregarPais(paisAux);
+
+            }
+
+            fileReader = new FileReader(rutaArchivo);
+            csvReader = new CSVReader(fileReader);
+            csvReader.readNext();
+            while ((linea = csvReader.readNext()) != null) {
+                ArrayList<Pais> frontera = new ArrayList<Pais>();
+                Pais paisAux = paises.get(linea[0]);
+                String[] listaFronteras = linea[2].replaceAll("^\"|\"$", "").split(",");
+
+                for (String pais : listaFronteras){
+                    frontera.add(paises.get(pais));
+                }
+
+
+                paisAux.agregarFrontera(frontera);
+
+
+            }
+        }
+        catch (Exception e) {
+            e.printStackTrace();
+        }
+
+
+    }
+
+    public void crearContinentes(){
+        ArrayList<Integer> bonusFichas = new ArrayList<>();
+        bonusFichas.add(5);
+        bonusFichas.add(3);
+        bonusFichas.add(7);
+        bonusFichas.add(3);
+        bonusFichas.add(5);
+        bonusFichas.add(2);
+        String[] nombresContinentes = {"America del Norte", "America del Sur", "Asia", "Africa", "Europa", "Oceania"};
+        for (int i = 0; i < (nombresContinentes.length); i++){
+            continentes.put(nombresContinentes[i], new Continente(nombresContinentes[i],bonusFichas.get(i)));
+        }
+    }
+
+    public void repartirPaises(ArrayList<Jugador> jugadores){
+
+        int cantidadPaises = paises.size();
+        int cantidadJugadores = jugadores.size();
+        Pais paisAux;
+        String[] keys = paises.keySet().toArray(new String[0]);
+        int i = 0;
+        int j = 0;
+        int jugActual;
+        while(i < cantidadPaises){
+            jugActual = j % cantidadJugadores;
+            paisAux = paises.get(keys[i]);
+            paisAux.asignarEjercito(jugadores.get(jugActual).getEjercito());
+            paisAux.agregarEjercito(1);
+            i++;
+            j++;
+        }
```

Comment:
se puede usar:
```
for(Pais pais: paises){
  ...
}
```
incluso pueden usar el forEach me parece

Reply:
Hola Edon, aca cuando queremos meter un for nos tira "foreach not applicable to type 'java.util.Hashtable<java.lang.String,edu.fiuba.algo3.modelo.distribuciondepaises.Pais>'" marcando "paises" en rojo